src/templates/CentralTemplateProvider.ts

```diff
@@ -123,12 +123,13 @@ export class CentralTemplateProvider implements Disposable {
         return await provider.getProjKey(context);
     }
 
-    private getCachedProvidersKey(language: ProjectLanguage, version: FuncVersion): string {
-        return language + version;
+    private getCachedProvidersKey(language: ProjectLanguage, languageModel: number | undefined, version: FuncVersion): string {
+        // NOTE: VS Code treats lack of a language model project setting as a 0, so treat undefined === 0.
+        return `${language}:${languageModel !== undefined ? languageModel : 0}:${version}`;
```

Comment:
```suggestion
        return `${language}:${languageModel ?? 0}:${version}`;
```
Nullish coalescing operator has the slight advantage of also catching `null`, in addition to `undefined` (even though in theory it shouldn't be happening...). Plus IMO it's a little cleaner but that's personal preference.

Reply:
Ok, I'll buy that.